src/base/emu-i386/simx86/cpu-emu.c

```diff
@@ -1115,7 +1115,9 @@ int e_vm86(void)
 #endif
   e_sigpa_count = 0;
   mode = ADDR16|DATA16; TheCPU.StackMask = 0x0000ffff;
-  TheCPU.mem_base = (uintptr_t)mem_base;
+  /* The simulator uses dosaddr_t throughout, the JIT adds mem_base
+     to the segment bases */
+  TheCPU.mem_base = CONFIG_CPUSIM ? 0 : (uintptr_t)mem_base;
```

Comment:
Maybe sim will then just stop using TheCPU.mem_base
if it doesn't need it?

Reply:
There is quite a bit of common code, macros, etc, and testing for CONFIG_CPUSIM in all of those is annoying (and more expensive then simply adding or subtracting 0)